### PR TITLE
Chore: (Docs)Adds Docs 2 documentation

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request, the documentation is updated to factor in the upcoming change to Docs and how to enable them in Storybook 7.0.

What was done:
- Created the snippets required for the documentation
- Updated the Toc file for accuracy
- Reworked the docs page based on the required changes
- Cleaned up the references to DocsPage from the docs
- Updated the DocsPage references in some of the Doc Blocks documentation

Some considerations regarding this pull request:
- The subcomponents usage was removed on purpose and will be added in a follow-up pr to feature in a more suitable place.
- Inline rendering was also removed on purpose, and similar to the above will be added as part of the pr mentioned above
- Doc Blocks are not updated yet as we don't have enough clarity on the changes. Once we have it the documentation will be updated accordingly.

### CODE REVIEW

File: docs/writing-docs/doc-block-canvas.md
Comment (tmeasday) [inline docs/writing-docs/doc-block-canvas.md:9]: I think these snippets refer to how the automatically generated docs works. Are we moving away from calling that "DocsPage" (I'm cool with that) -- if so we still probably need to differentiate it from docs in general though (like these doc blocks only ever work in "Docs" so what does "Working with Docs" mean?)

The tag is still called `docsPage` although we could change this. Personally I've been using "autodocs" in my head a bit recently. Maybe we should call it "Automatic Docs" in our documentation?

So this section would be something like "Canvas in Automatic Docs"? WDYT @shilman?

File: docs/writing-docs/docs-page.md
Comment (tmeasday) [inline docs/writing-docs/docs-page.md:11]: I think we should make it clear that the final step is optional, and isn't required.

File: docs/writing-docs/docs-page.md
Comment (tmeasday) [inline docs/writing-docs/docs-page.md:15]:
```suggestion
To enable auto-generated documentation for your stories, you'll need to add the `tags` configuration property to the CSF file's default export. For example:
```

File: docs/writing-docs/docs-page.md
Comment (tmeasday) [inline docs/writing-docs/docs-page.md:39]: What does "positioned at the root level" mean here?

File: docs/writing-docs/docs-page.md
Comment (tmeasday) [inline docs/writing-docs/docs-page.md:59]: We should also mention the option `'automatic'` which means that every component gets an automatic docs page entry without needing to add the tag.

File: docs/writing-docs/docs-page.md
Comment (tmeasday) [inline docs/writing-docs/docs-page.md:98]: This sort of implies the name needs to be the same, rather than just that being a suggestion. What *needs* to happen is referencing the CSF file in a `<Meta of={}/>`

File: docs/writing-docs/docs-page.md
Comment (tmeasday) [inline docs/writing-docs/docs-page.md:110]: We are actually going to add a warning if you set the tag but then override the auto doc with an MDX file: https://linear.app/chromaui/issue/SB-1032/warn-if-setting-tags-[docspage]-and-then-overriding-with-meta-of=

So I'm wondering if this should explicitly say you should drop the tag if you create an MDX file.

**Review by tmeasday — COMMENTED** [review]
Some feedback!

File: docs/writing-docs/doc-block-canvas.md
Comment (jonniebigodes) [reply docs/writing-docs/doc-block-canvas.md:9]: @tmeasday I'm actually been thinking the same about the naming. Autodocs has been going back and forward in my head as well. If @shilman is on board with it as well I'm more than glad in making the change.

Comment (tmeasday) [reply docs/writing-docs/doc-block-canvas.md:9]: I think if we want to go with that name we should change the name of the option and tag also.

**Review by ShaunEvening — APPROVED** [review]
This looks awesome, Joao!

File: docs/writing-docs/docs-page.md
Comment (tmeasday) [inline docs/writing-docs/docs-page.md:80]: I realise I misspoke before. If you are using autotitle, then maming the file the same as the CSF file is also a valid approach for sure. I'm not sure if we should mention that or not, WDYT @shilman ?

Comment (jonniebigodes) [reply docs/writing-docs/docs-page.md:80]: No problem @tmeasday, I'll defer to @shilman and adjust accordingly :)

File: docs/writing-docs/doc-block-canvas.md
Comment (shilman) [reply docs/writing-docs/doc-block-canvas.md:9]: I'm pretty neutral on the change. Fine with it if that's the consensus. I agree with @tmeasday that we should make it consistent if we do

Comment (kylegach) [reply docs/writing-docs/doc-block-canvas.md:9]: I think we need to be careful here about replacing "DocsPage" with just "Docs". In my mental model, both "DocsPage" and MDX fall under the umbrella of "Docs".

I like the suggestion of replacing "DocsPage" with "Automatic Docs" (`autoDocs` in config). Especially because this differentiates it from `parameters.docs.page`.

File: docs/faq.md
Comment (kylegach) [inline docs/faq.md:221]: Here's an example of what [this comment](https://github.com/storybookjs/storybook/pull/20233#discussion_r1049172744) was referring to.

File: docs/snippets/common/storybook-auto-docs-full-config.js.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-auto-docs-full-config.js.mdx:19]: Given the full table below, this seems like a more realistic/useful example?

```suggestion
    docsPage: 'automatic',
    defaultName: 'Documentation',
```

File: docs/writing-docs/docs-page.md
Comment (kylegach) [inline docs/writing-docs/docs-page.md:60]: I don't understand the difference between `docs.enabled = false` and `docs.docsPage = false`.

File: docs/snippets/common/storybook-preview-auto-docs-template.mdx.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-preview-auto-docs-template.mdx.mdx:2]: To match the snippet that references it.

```suggestion
{/* DocumentationTemplate.mdx */}
```

File: docs/snippets/common/storybook-preview-auto-docs-template.mdx.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-preview-auto-docs-template.mdx.mdx:4]: This seems like a weird example to use for a project-level template?

What if we used this to show the default template, which would hint at how to use various Doc Blocks?

File: docs/writing-docs/docs-page.md
Comment (kylegach) [inline docs/writing-docs/docs-page.md:65]: > written in MDX

Does this mean that `parameters.docs.page` no longer accepts a React component?

File: docs/snippets/common/storybook-auto-docs-main-mdx-config.js.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-auto-docs-main-mdx-config.js.mdx:5]: More helpful?

```suggestion
  stories: [
    '../src/**/*.mdx', // 👈 add this, to match your project's structure
    '../src/**/*.stories.@(js|ts)'
  ],
```

File: docs/writing-docs/docs-page.md
Comment (kylegach) [inline docs/writing-docs/docs-page.md:86]: > To enable custom documentation for your stories with this format, start by updating your Storybook configuration file

Doesn't _any_ use of MDX in SB use MDX 2? This seems a little misleading, and perhaps this intro to MDX should be higher up this document?

File: docs/snippets/common/storybook-auto-docs-starter-example.mdx.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-auto-docs-starter-example.mdx.mdx:10]: More realistic.

```suggestion
# Button
```

File: docs/snippets/common/storybook-auto-docs-mdx-file.mdx.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-auto-docs-mdx-file.mdx.mdx:11]:
```suggestion
# Button
```

File: docs/snippets/common/storybook-auto-docs-mdx-file.mdx.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-auto-docs-mdx-file.mdx.mdx:25]:
```suggestion
# Page
```

File: docs/snippets/common/storybook-auto-docs-mdx-file.mdx.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-auto-docs-mdx-file.mdx.mdx:1]: I don't think it needs to block this PR, but I find this a confusing example of combining components into one doc. How is Button related to Page? How about List and ListItem?

File: docs/snippets/common/storybook-preview-auto-docs-custom-docs-container.js.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-preview-auto-docs-custom-docs-container.js.mdx:6]: It'd be great to demo at least some of the useful `props` you're given.

E.g., I've personally used this feature to avoid defining the title (pre "auto title") in both the CSF meta and the MDX (as an h1).

File: docs/snippets/common/storybook-preview-auto-docs-custom-docs-container.js.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-preview-auto-docs-custom-docs-container.js.mdx:28]: `ExampleContainer` is already a React component, so need to wrap it in another one that just passes along all props.

```suggestion
    container: ExampleContainer,
```

File: docs/writing-docs/docs-page.md
Comment (kylegach) [inline docs/writing-docs/docs-page.md:145]: I think it's important to mention that we strongly recommend rendering the SB DocsContainer inside their custom one, because it provides context used by various Docs Blocks.

File: docs/snippets/common/storybook-auto-docs-override-mdx-container.js.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-auto-docs-override-mdx-container.js.mdx:20]: Probably good to not assume they read the prior section and just show how to apply it here too:

```suggestion
export const parameters = {
  actions: { argTypesRegex: '^on[A-Z].*' },
  controls: {
    matchers: {
      color: /(background|color)$/i,
      date: /Date$/,
    },
  },
  docs: {
    container: MyDocsContainer,
  },
};
```
```

File: docs/snippets/common/storybook-auto-docs-override-mdx-container.js.mdx
Comment (kylegach) [inline docs/snippets/common/storybook-auto-docs-override-mdx-container.js.mdx:19]:
```suggestion
export const MyDocsContainer = (props) => (
  <MDXProvider
    components={{
      h1: DesignSystem.H1,
      h2: DesignSystem.H2,
    }}
  >
    <DocsContainer {...props} />
  </MDXProvider>
);

export const parameters = {
  actions: { argTypesRegex: '^on[A-Z].*' },
  controls: {
    matchers: {
      color: /(background|color)$/i,
      date: /Date$/,
    },
  },
  docs: {
    container: MyDocsContainer,
  },
};
```

File: docs/writing-docs/introduction.md
Comment (kylegach) [inline docs/writing-docs/introduction.md:9]: This image doesn't actually show Docs (aside from the orange icons in the sidebar).

File: docs/writing-docs/introduction.md
Comment (kylegach) [inline docs/writing-docs/introduction.md:11]: > If you're including Storybook in your project for the [first time](../get-started/install.md) or migrating from a previous version

I don't think this detail is necessary, now that addon-docs has been a default part of SB for so long.

File: docs/writing-docs/doc-block-story.md
Comment (kylegach) [inline docs/writing-docs/doc-block-story.md:24]: Given that we're removing all docs for writing stories in MDX (with a link back to older docs)...

Wouldn't all of these annotation props (`args`, `decorators`, `loaders`, `parameters`, `play`) go away as well, because they're only defined in CSF now?

Happy to update this as part of my MDX 2 PR (coming soon!); just let me know!

File: docs/writing-docs/doc-block-argstable.md
Comment (kylegach) [inline docs/writing-docs/doc-block-argstable.md:38]: @JReinhold — I cannot comment directly on it, but for the props table below, is `of` correct? I figured it would match `<Meta>` and accept the CSF meta. And, if that's correct, is `story` deprecated (or will be soon)?

```md
| Option  | Description                                                                                         |
| ------- | --------------------------------------------------------------------------------------------------- |
| `of`    | Infers the args table from the component. <br/> `<ArgsTable of={MyComponent} />`                    |
| `story` | Infers the args table based on a story. <br/> `<ArgsTable story="example-mycomponent--my-story" />` |
```

Comment (kylegach) [reply docs/writing-docs/doc-block-argstable.md:38]: Would appreciate you weighing in on any other Docs Blocks API changes, while you're here. 🙂

File: docs/writing-docs/introduction.md
Comment (jonniebigodes) [reply docs/writing-docs/introduction.md:9]: I know, and I went with this one as a placeholder for two reasons:
1- Show off the unified UI
2- We actually don't have that many examples showcasing the feature yet. Once we do this one will be updated.

File: docs/writing-docs/doc-block-story.md
Comment (jonniebigodes) [reply docs/writing-docs/doc-block-story.md:24]: Indeed they should, but this is a different conversation altogether. One that I'm going to have with @JReinhold to get some clarification. I've mentioned this in the pr description

File: docs/writing-docs/docs-page.md
Comment (kylegach) [reply docs/writing-docs/docs-page.md:60]: Chatted about this and now understand the distinction, so:

```suggestion
| `enabled`     | Toggles support for all documentation pages <br/> `docs: { enabled:false }`                 |
| `docsPage`    | Disables auto-generated documentation pages created via `tags` <br/> `docs: { docsPage: false }`       |
| `automatic`   | Enables auto-generated documentation pages for every component <br/> `docs: { docsPage: 'automatic' }` |
```

I'm still not confident we need both options, but 🤷‍♂️.

File: docs/snippets/common/storybook-preview-auto-docs-template.mdx.mdx
Comment (kylegach) [reply docs/snippets/common/storybook-preview-auto-docs-template.mdx.mdx:4]: Rewrite this in MDX: https://github.com/storybookjs/storybook/blob/next/code/ui/blocks/src/blocks/DocsPage.tsx

File: docs/writing-docs/docs-page.md
Comment (kylegach) [reply docs/writing-docs/docs-page.md:65]:
```suggestion
To replace the default documentation template used by Storybook, you can create your own template written in MDX and update your UI configuration file (i.e., `.storybook/preview.js`) to point to it.
```

File: docs/snippets/common/storybook-preview-auto-docs-custom-docs-container.js.mdx
Comment (kylegach) [reply docs/snippets/common/storybook-preview-auto-docs-custom-docs-container.js.mdx:6]: Or, alternatively, show how you can access what's in the context via the hook?

Comment (kylegach) [reply docs/snippets/common/storybook-preview-auto-docs-custom-docs-container.js.mdx:6]: I'll provide an example in a bit...

Comment (kylegach) [reply docs/snippets/common/storybook-preview-auto-docs-custom-docs-container.js.mdx:6]: Decided it was best to not to be prescriptive when providing an example:

```jsx
import * as React from "react";
import { DocsContainer } from "@storybook/blocks";

const ExampleContainer = ({ children, ...props }) => {
  console.log("DocsContainer props", props);

  return <DocsContainer {...props}>{children}</DocsContainer>;
}

export const parameters = {
  actions: { argTypesRegex: "^on[A-Z].*" },
  controls: {
    matchers: {
      color: /(background|color)$/i,
      date: /Date$/,
    },
  },
  docs: {
    container: ExampleContainer,
  },
};
```

File: docs/writing-docs/docs-page.md
Comment (shilman) [reply docs/writing-docs/docs-page.md:80]: I don't think there's any auto-title for MDX

**Pull request merged**
